nntrainer/tensor/tensor.cpp

```diff
@@ -750,6 +752,49 @@ Tensor Tensor::getSharedDataTensor(const TensorDim dim_, unsigned int offset,
   return ret;
 }
 
+std::vector<Tensor> Tensor::split(unsigned num_size, int axis) {
+  if (axis == -1) {
+    axis = 3;
+  }
+  NNTR_THROW_IF(!(0 <= axis && axis < 4), std::invalid_argument)
+    << "cannot split axis of axis: " << axis;
+
+  NNTR_THROW_IF(dim.getTensorDim(axis) % num_size != 0, std::invalid_argument)
+    << "axis is not divisible by num_size, axis: " << axis
+    << " num size: " << num_size;
+
+  auto ret_dim = dim;
+  auto new_dim = dim.getTensorDim(axis) / num_size;
+  ret_dim.setTensorDim(axis, new_dim);
+
+  auto iter_value = [this, &ret_dim](std::array<unsigned, 4> &loc) {
+    auto value = getValue(loc[0], loc[1], loc[2], loc[3]);
+    for (int i = 3; i >= 0; --i) {
```

Comment:
I would highly recommend you to refer the split layer implementation, where changes the shape the tensor to be split to not only simplify the computation but also makes it quite efficient.

Reply:
this is return 1 value at a time with this operation, using getValue() - this is needed only along the axis for the split right and not along all the dimensions?

Reply:
The strategy is good, for now I'll stay with naiver implementation and improve later, as this is only aimed to be used for non-critical path (weight saving)
I have greater plan to implement a iterator something like numpy.nditer('external_loop') to generalize the behavior

Reply:
alright, but these methods should work without iterator too for non-strided tensors as they will be more efficient.

Reply:
Yeah, as per offline discussion, the iterator should cover the scenario with an option.